src/bp-forums/common/shortcodes.php

```diff
@@ -694,8 +694,14 @@ public function display_view( $attr, $content = '' ) {
 		 */
 		public function display_search_form() {
 
+			global $post;
+			$has_bbp_search_form_shortcode = false;
+			if ( ! empty( $post->post_content ) && has_shortcode( $post->post_content, 'bbp-search-form' ) ) {
+				$has_bbp_search_form_shortcode = true;
+			}
+
 			// Bail if search is disabled.
-			if ( ! bbp_allow_search() ) {
+			if ( ! $has_bbp_search_form_shortcode && ! bbp_allow_search() ) {
```

Comment:
Revert the code changes to allow the search for the shortcode.
We have the option to enable/disabled search. https://prnt.sc/toNUTFeVrvjK